### PR TITLE
PD Fixups: Remove old `Labware` flow type

### LINKED ISSUE #3052: PD Fixup Round: deprecate old `Labware` type
deprecate old `Labware` type - remove completely. Should always be LabwareEntities or DisplayLabware (outside of `step-generation`, which has its own internal type for its different use case)

remove `getLabwareById` selector and its dependencies

Right now, `step-form` selectors are re-implementing `labware-ingred` selectors as a hack to avoid circular dependency. Instead, we should clearly delineate the different concerns, neither of these "atoms" (step-form vs labware-ingred) should have selectors that rely on state from the other.

### PR DESCRIPTION
**Pull request by IanLondon**
## overview

Backstory for future reference: we're been moving to split labware information into two data types:
- `LabwareEntity` is a responsibility of `step-forms/`. A `LabwareEntity` has a `type` (eg `96-flat`), and that's it for now and the foreseeable future.
- `DisplayLabware` is the responsibility of `labware-ingred/`. (Side note: we plan to eventually move labware-ingred into `ui/` for the ui concerns and `step-forms/` for the ingredient stuff

Closes #3052

## changelog

* remove `Labware` type and all uses. Instead, use `LabwareEntity` or `DisplayLabware`.
* prefer `getLabwareTypesById` to `getLabwareEntities`, unless there's a good reason to get the whole entities map
* move getLabwareNicknamesById, getDisposalLabwareOptions, getLabwareOptions to new `ui/labware/selectors.js`
* refactor StepItem props to take labware nicknames and labware types separately, refactor props of the StepItem child components to take a flat props object of strings, instead of `Labware`

## review requests

There should be no user-facing changes from edge

Things touched to test:
- [ ] Step items (headers and tooltips related to labware names)
- [ ] Well highlighting (via steps & substeps)
- [ ] Labware naming & renaming (including duplication)
- [ ] Well selection modal functionality
- [ ] Final deck state view
- [ ] Title bar shows labware correctly in all the places it should (add liquids modal, final deck state view, ??)

### CODE REVIEW

File: protocol-designer/src/containers/ConnectedTitleBar.js [hunk 1]
```diff
@@ -43,18 +44,19 @@ function TitleWithIcon (props: TitleWithIconProps) {
...
   const selectedTerminalId = stepsSelectors.getSelectedTerminalItemId(state)
-  const labware = selectedLabware
-  const labwareNames = stepFormSelectors.getLabwareNicknamesById(state)
-  const labwareNickname = labware && labware.id && labwareNames[labware.id]
+  const labwareNames = uiLabwareSelectors.getLabwareNicknamesById(state)
   const drilledDownLabwareId = labwareIngredSelectors.getDrillDownLabwareId(state)
-  const liquidPlacementMode = Boolean(selectedLabware)
   const wellSelectionLabwareKey = stepsSelectors.getWellSelectionLabwareKey(state)
 
+  const labwareNickname = selectedLabwareId != null && labwareNames[selectedLabwareId]
```
Comment (IanLondon) [inline protocol-designer/src/containers/ConnectedTitleBar.js]: TODO, these are weird fallbacks

File: protocol-designer/src/top-selectors/well-contents/getWellContentsAllLabware.js [hunk 2]
```diff
@@ -56,29 +56,32 @@ const _getWellContents = (
...
+  (labwareTypesById, liquidsByLabware, selectedLabwareId, selectedWells, highlightedWells) => {
+    // TODO: Ian 2019-02-14 weird flow error without explicit Array<string> annotation
+    const allLabwareIds: Array<string> = Object.keys(labwareTypesById)
 
-    return allLabwareIds.reduce((acc: {[labwareId: string]: ContentsByWell | null}, labwareId: string) => {
+    return allLabwareIds.reduce((acc: WellContentsByLabware, labwareId: string) => {
       const liquidsForLabware = liquidsByLabware[labwareId]
-      const isSelectedLabware = selectedLabware && (selectedLabware.id === labwareId)
+      const isSelectedLabware = selectedLabwareId === labwareId
+      console.log({isSelectedLabware, labwareId, selectedLabwareId})
```
Comment (IanLondon) [inline protocol-designer/src/top-selectors/well-contents/getWellContentsAllLabware.js]: TODO stray log

File: protocol-designer/src/containers/LabwareContainer.js [hunk 3]
```diff
@@ -91,41 +110,43 @@ function mapStateToProps (state: BaseState, ownProps: OP): SP {
...
+    // TODO: Ian 2019-02-14 this fallback to '' is weird
+    containerType: labwareType || '',
+    containerId: labwareId,
   }
 }
 
 function mergeProps (stateProps: SP, dispatchProps: {dispatch: Dispatch<*>}, ownProps: OP): Props {
   const {slot} = ownProps
   const {dispatch} = dispatchProps
   const {containerId, containerName, containerType} = stateProps
```
Comment (b-cooper) [inline protocol-designer/src/containers/LabwareContainer.js]: can `containerName` and `containerType` be removed now? or atleast renamed to `labwareXXX` instead?

File: protocol-designer/src/labware-ingred/selectors.js [hunk 4]
```diff
@@ -17,14 +17,15 @@ import type {
...
+import type {BaseState, Options, Selector} from './../types'
 
-type Selector<T> = (RootSlice) => T
+// TODO: Ian 2019-02-15 no RootSlice, use BaseState
 type RootSlice = {labwareIngred: RootState}
 
 const rootSelector = (state: RootSlice): RootState => state.labwareIngred
 
-const getLabwareNameInfo = createSelector(
+// NOTE: not intended for UI use! Use getLabwareNicknamesById for the string.
```
Comment (b-cooper) [inline protocol-designer/src/labware-ingred/selectors.js]: Should this NOTE just be a TODO to rename the selector to something more explanatory?

File: protocol-designer/src/well-selection/reducers.js [hunk 5]
```diff
@@ -39,27 +38,17 @@ const selectedWells = handleActions({
   }),
   // Actions that cause "deselect everything" behavior:
   CLOSE_INGREDIENT_SELECTOR: () => selectedWellsInitialState,
-  CLOSE_WELL_SELECTION_MODAL: () => selectedWellsInitialState,
   DESELECT_ALL_WELLS: () => selectedWellsInitialState,
   REMOVE_WELLS_CONTENTS: () => selectedWellsInitialState,
   SET_WELL_CONTENTS: () => selectedWellsInitialState,
 }, selectedWellsInitialState)
 
-// TODO: BC 2018-11-01 unused, remove and all references to the actions
```
Comment (b-cooper) [inline protocol-designer/src/well-selection/reducers.js]: 💣

File: protocol-designer/src/containers/LabwareContainer.js (hunk 3 above)
Comment (IanLondon) [reply protocol-designer/src/containers/LabwareContainer.js]: I didn't want to expand the diff. We should bump up priority of #1773 and get rid of all `containerXxx` together :bomb:

**Comment by IanLondon** [comment]
Talked in persion: #3100 will address comments about overly-general nullsy checks

**Pull request merged**